lib/rules/no-mutating-props.js

```diff
@@ -423,12 +478,22 @@ module.exports = {
             return
           }
 
+          const propsInfo = /** @type {PropsInfo} */ (
+            propsMap.get(vueObjectData.object)
+          )
+
           const nodes = utils.getMemberChaining(node)
           const first = nodes[0]
           let name
-          if (isVmReference(first)) {
+          if (isVmReference(first) && first.name !== propsInfo.name) {
```

Comment:
It looks like the following cases are no longer reported (without new option):

```vue
<script setup>
  const props = defineProps()
</script>

<template>
  <input v-model="props[foo]" >
</template>
```

I think we should branch like this:

```js
          if (isVmReference(first)) {
            if (first.name === propsInfo.name) {
              // props variable
              if (shallowOnly && nodes.length > 2) {
                return
              }
              name =
                (nodes[1] && utils.getStaticPropertyName(nodes[1])) ||
                first.name
            } else {
              if (shallowOnly && nodes.length > 1) {
                return
              }
              name = first.name
              if (!name || !propsInfo.set.has(name)) {
                return
              }
            }
          } else if (first.type === 'ThisExpression') {
            if (shallowOnly && nodes.length > 2) {
              return
            }
            const mem = nodes[1]
            if (!mem) {
              return
            }
            name = utils.getStaticPropertyName(mem)
            if (!name || !propsInfo.set.has(name)) {
              return
            }
          } else {
            return
          }
          report(node, name)
```

Also, could you add this test case as well?

Reply:
Of course, I have fixed it and added related tests.